sips/sip-x/sip-x-authentication-protocol.md

```diff
@@ -0,0 +1,397 @@
+
+In addition to the public key, more information can be shared like email address or profile pictures. Some data can be shared publicly, other only with the application. In particular, a private key is derived by the authenticator that is specific to the application and to the user. This private key can be used by the application for example to access decentralized storage detailed in the response or sign messages in the name of the user of the application.
+
+# Specification
+
+The basic flow of the authentication between the application and the authenticator (aka wallet or agent of the user) is as follows:
+
+1. Application creates app transit private key, signs an auth request with that key and sends the request to the Authenticator.
+2. In the Authenticator, User authorizes sharing of public key, Authenticator derives app private key from request and updates the user's public profile if required.
+3. Authenticator creates response with authorized data and sends response to the Application.
+4. Application verifies signature against the app transit private key.
+
+https://cogarius.medium.com/blockstack-world-tour-brussels-social-dapp-workshop-fb0ef887b55f
+
+Requests and responses are signed json web tokens (JWT) following standard [RFC 7519](https://tools.ietf.org/html/rfc7519).
+
+The header of any JWT must be
+
+```
+{
+  "typ": "JWT",
+  "alg": "ES256K"
+}
+```
+
+The payloads for the authentication request and the response are specified in the following sections.
+
+## Authentication Request
+
+The authentication request is a JWT created by the application. It is signed by a private key that must be freshly generated. The key is called app transit key.
+
+### Signed JWT
+
+The payload must contain the following claims:
+
+| Claim name             | Type   | Description                                                                                                                               |
+| ---------------------- | ------ | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| jti                    | string | As defined in RFC7519                                                                                                                     |
+| iat                    | string | As defined in RFC7519                                                                                                                     |
+| exp                    | string | As defined in RFC7519                                                                                                                     |
+| iss                    | string | Decentralized identifier defined in DID specification representing the user's account. See Appendix C for list of well-known DID methods. |
```

Comment:
This seems to be a combination of both an authResponsePayload and an authRequestPayload. How can the authRequest include the DID of an address its not yet aware of?

Reply:
It is a did representation of the public transit key. I have updated to description.